pkg/datapath/loader/base.go

```diff
@@ -511,3 +513,63 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 
 	return nil
 }
+
+// OnUpdateCiliumNode is only called if running with IPAM ENI and IPsec. In
+// that case, we need to rerun reinitializeIPSec whenever new ENIs are added to
+// the local node, to attach our BPF program for IPsec decryption to the new
+// interfaces.
+func (l *Loader) OnUpdateCiliumNode(oldNode, newNode *ciliumv2.CiliumNode, _ *lock.StoppableWaitGroup) error {
+	if !k8s.IsLocalCiliumNode(newNode) {
+		return nil
+	}
+
+	log.Debug("Checking for new ENIs on local node")
+
+	newENIs := make([]string, 0)
+	for newENI := range newNode.Status.ENI.ENIs {
+		existed := false
+		for oldENI := range oldNode.Status.ENI.ENIs {
+			if newENI == oldENI {
+				existed = true
+				break
+			}
+		}
+		if !existed {
+			newENIs = append(newENIs, newENI)
+		}
+	}
+
+	if len(newENIs) == 0 {
+		return nil
+	}
+
+	// ... and we can then trigger loading of BPF programs to those interfaces.
+	log.WithField(logfields.NewENIs, newENIs).Info("New ENIs detected")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// If we have any new ENIs, we first need to wait for the corresponding
+	// interfaces to show up...
+	missingENIByMac, err := ipam.WaitForENIInterfaces(oldNode, newNode)
+	if err != nil {
+		log.WithError(err).WithFields(logrus.Fields{
+			logfields.MissingENIs: missingENIByMac,
+		}).Error("Failed while waiting for ENIs to be attached")
+		return err
+	}
+
+	return l.reinitializeIPSec(ctx)
```

Comment:
Doesn't this need to be called mutually exclusive with `l.Reinitialize`? Otherwise both will write to shared global state without synchronization.